src/main/java/data/JsonConverter.java

```diff
@@ -0,0 +1,55 @@
+
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package data;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.json.JSONObject;
+
+/** A class that creates a config file from a Vendor object. */
+public class JsonConverter {
+  private static final String FILE_PATH_BASE = "data/";
+  
+  public File createFile() {
```

Comment:
The design doc calls out a FileUtil class for writing a file. My impression was that, this JsonConverter class would handle serializing Account/Customer objects to a JSON string. Then, FileUtil would call JsonConverter to get the string representation of an object, and handle the file IO necessary to write this to the file system.

I see that the Json Converter section of the doc expects FileUtil to call this method to write the file. So I am okay with this approach, as it seems the backend components are on the same page.

I think we can leverage the Customer/Account classes more here, though. In a future revision, I would expect the "practiceFile" portion of the file name to be provided though a parameter, along with a Customer object containing the object to write. I think it would be useful to have .toJson() methods on both the Customer and Account objects.

Then, this method calls Customer.toJson(), which writes the customer identifiers and calls Account.toJson() on each account object to build the final JSON representation. This whole string (or JSONObject) is returned from Customer.toJson() and written from this method.

Also, is this class going to handle reading from the filesystem? We may want to stub out a readFile method that reads from "practiceFile" and returns a JSONObject/Customer/Account.

Reply:
The idea was that the FileUtil class calls the updateFile(Vendor) method in the JsonConverter class. 

In the future, a filePath will be created using the BasePath(data/) and the vendorID. Adding a .toJson() to the Vendor and Account classes is a good idea, which I'll make in the future. 

Currently, I don't plan on having a readfile in the JsonConverter. The plan was to just write files to the filesystem.

Reply:
I'll let the rest of the team know that I will add readFile functionality to the JsonConverter class on the next iteration.